pkg/action/env.container.go

```diff
@@ -234,6 +234,7 @@ func (c *containerEnv) Execute(ctx context.Context, a *Action) (err error) {
 	log.Info(msg)
 	if status != 0 {
 		err = RunStatusError{code: status, msg: msg}
+		return err
```

Comment:
It doesn't seem to be a correct way to fix the issue. By returning here, you are skipping other important defers.
I think the better solution is the following:
```go 
// Change this
err = c.imageRemove(ctx, a)
// to this
errImg := c.imageRemove(ctx, a)
// or this
err := c.imageRemove(ctx, a)
```

In this defer you are overriding the global error. Usually to avoid shadowing, devs use `errImg`